src/touchid/TouchID.h

```diff
@@ -21,22 +21,17 @@ class TouchID
     bool storeKey(const QString& databasePath, const QByteArray& passwordKey);
     bool getKey(const QString& databasePath, QByteArray& passwordKey) const;
     bool containsKey(const QString& databasePath) const;
-    bool isAvailable();
     void reset(const QString& databasePath = "");
 
+    static bool isAvailable();
+    static bool isWatchAvailable();
+    static bool isTouchIdAvailable();
```

Comment:
`isAvailable()` should not be static, since the same is needed in Windows Hello (see [source](https://github.com/keepassxreboot/keepassxc/blob/develop/src/winhello/WindowsHello.h)) and later in the Linux equivalent. Once they're abstracted to a common interface as was planned, `isAvailable()` needs to be in that interface. It would be hardware and OS agnostic from that point of view, with the details handled by each implementation.

`isWatchAvailable()` is one of those implementation details that are specific to macOS. Probably shouldn't be public.

Reply:
> From the application side, it shouldn't care whether the authentication happens with the watch or a fingerprint reader. It just needs to store and read the key, when that's possible.

I think it's always better to communicate as much information to the user as possible, otherwise it's really hard to debug the authentication issues. At the moment I, as a Mac user have no clue why quick unlock doesn't work, which methods are available, is it a OS problem or KeePassXC problem etc. 

1Password does a bit better job. They explicitly communicate that unlock is impossible with the laptop lid being  closed, watch being out of range etc. 🤷‍♂️

Reply:
I agree, if the specific errors are made available by the api then we should show them. Windows hello doesn't provide much of anything.

Reply:
That's usually what exceptions are for. As long as the calling code can catch the exception and display the error message, that part is independent from the class interface.

What I meant by "the application shouldn't care" is that it shouldn't need to explicitly check which hardware is available and use different code paths for each case. It would just call `getKey()` (or whatever), and let the implementation worry about the details. Then catch any error and display the message (also without worrying about what it says).